metrics/formulaCalculator.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"math"
+)
+
+type Factor struct {
+	Weight       float64
+	Value        float64
+	AllOrNothing bool
+}
+
+func ComputeNetScore(fs []Factor) float64 {
+	var sum float64
+
+	for _, f := range fs {
+		if f.AllOrNothing {
+			if f.Value == 0 {
+				sum = 0
+				break
+			} else {
+				continue
```

Comment:
this else feels unnecessary

Reply:
yeah idk why that's in there. I'll remove it

Reply:
turns out this is needed. Otherwise scores that are "all or nothing" will be added to the sum, but that is not correct. They should either leave the score as it was or set it to 0